excellent/functions/functions.go

```diff
@@ -1719,6 +1720,13 @@ func Length(env utils.Environment, value types.XValue) types.XValue {
 		return types.NewXNumberFromInt(lengthable.Length())
 	}
 
+	// or reducable to something with length
+	value = types.Reduce(env, value)
+	lengthable, isLengthable = value.(types.XLengthable)
```

Comment:
can value be nil?

Reply:
calling `types.Reduce(..)` instead of `value.Reduce()` for that reason because it handles `nil`.. but now you've got me worried about `value.(types.XLengthable)` returning an interface to a nil...

Reply:
Ya, that was my question there.

Reply:
Sounds like for interfaces, nil is never true.

Reply:
I think if you passed it an `XValue` interface to a `nil` that implemented `XLengthable`, then `isLengthable` would be true - the same problem we fixed with `@parent.contact`.. tho in this case `types.Reduce(..)` returns a real `nil` if you pass it an interface to a `nil`

Reply:
I fee l like we really need to figure out a strategy for this.. feels like we must be doing something wrong or approaching things wrong to get us in a state where it is so subtle. Just smells like endless bug squashing.

Reply:
maybe that strategy should be to never create a non-nil interface to a nil.. I need to dig into this more and figure out where they're coming from in the first place.